beat/heart.py

```diff
@@ -3667,9 +3628,10 @@ def get_waveform_mapping(
                     logger.warn(
                         'No response for target %s in '
                         'the collection!' % str(trace_id))
+            new_targets.append(target)
 
         targets = utility.weed_targets(
-            targets, self.stations, discard_targets=discard_targets)
+            new_targets, self.stations, discard_targets=discard_targets)
```

Comment:
what do we need new_targets for?

Reply:
because after for loop, targets get back to their previous version, which is BasicTarget! I don't know how to make it permanent!

Reply:
ah sure yes you must not append the domain targets to the basic targets..., please call new targets domain_targets